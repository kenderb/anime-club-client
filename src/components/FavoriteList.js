import React, { useState } from 'react';
import PropType from 'prop-types';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import './Dashboard.style.css';
import AnimeCard from './AnimeCard';

const FavoriteList = ({ favorites, user, animes }) => {
  const myfavorites = animes.filter(anime => favorites.includes(anime.id));
  const [currentNumber, setCurrentUser] = useState(1);
  const handleCurrentSelect = id => {
    setCurrentUser(id);
  };
  if (!user.loggedIn) return <Redirect to="/" />;

  return (
    <div>
      <h1 className="dashboard-title">
        My favorite List:
      </h1>
      <ul className="anime-wrapper d-flex">
        {
        myfavorites.map(anime => (
          <AnimeCard
            key={anime.id}
            anime={anime}
            onClickImage={() => handleCurrentSelect(anime.id)}
          />
        ))
      }
      </ul>
      <div className="anime-count-container">
        {currentNumber}
        /
        {myfavorites.length}
      </div>
    </div>
  );
};

const mapStateToprops = state => ({
  favorites: state.favorites,
  user: state.user,
  animes: state.animes,
});

FavoriteList.propTypes = {
  favorites: PropType.instanceOf(Array).isRequired,
  animes: PropType.instanceOf(Array).isRequired,
  user: PropType.instanceOf(Object).isRequired,
};

export default connect(mapStateToprops)(FavoriteList);
