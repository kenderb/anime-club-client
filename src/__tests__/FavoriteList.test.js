import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import { createStore, applyMiddleware } from 'redux';
import { BrowserRouter } from 'react-router-dom';
import combineReducers from '../reducers';
import FavoriteList from '../components/FavoriteList';

test('Registration component renders correctly', () => {
  const store = createStore(combineReducers, applyMiddleware(thunk));
  const tree = create(
    <Provider store={store}>
      <BrowserRouter>
        <FavoriteList />
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
