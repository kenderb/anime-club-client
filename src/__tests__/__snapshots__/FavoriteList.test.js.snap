// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration component renders correctly 1`] = `null`;
